sklearn/cluster/_spectral.py

```diff
@@ -191,6 +190,48 @@ def discretize(
     return labels
 
 
+def _spectral_clustering(
```

Comment:
At this point, should the body of `_spectral_clustering` be in `SpectralClustering.fit`?

Reply:
I am fine moving the code in `fit` because `spectral_clustering` was still doing some validation. In the past we used sometimes the pattern to still have a private function that does not validate the input data.